### PR TITLE
@alloy => Re-structure the app loadup to not load the UI when doing background stuff 

### PR DESCRIPTION
**Pull request by orta**
OK, so there's some extra holdovers too. Will comment inline.

### CODE REVIEW

File: Artsy.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1794,16 +1800,6 @@
...
 		};
-		342F929D496BA3C99B4DB11A /* State Restoration */ = {
-			isa = PBXGroup;
-			children = (
-				342F997FDCA4513495F1E949 /* UIViewController+ARStateRestoration.h */,
-				342F992BB6F28B5BE6477E3F /* UIViewController+ARStateRestoration.m */,
-			);
-			name = "State Restoration";
-			path = State_Restoration;
-			sourceTree = "<group>";
```
Comment (orta) [inline Artsy.xcodeproj/project.pbxproj]: these were not used

File: Artsy.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1702,7 +1707,7 @@
 			files = (
 				60476F4C1AD87DB200670502 /* UIKit.framework in Frameworks */,
 				B3ECE8291819D1E6009F5C5B /* Foundation.framework in Frameworks */,
-				1AE74AC3E4428F2CD96DD625 /* libPods-Artsy Tests.a in Frameworks */,
+				0075C66DE3189142CD863793 /* libPods-Artsy Tests.a in Frameworks */,
```
Comment (orta) [inline Artsy.xcodeproj/project.pbxproj]: in going back from frameworks I `pod Deintegrated`  and we got new UUIDS

File: Artsy.xcodeproj/xcshareddata/xcschemes/Artsy.xcscheme [hunk 3]
```diff
@@ -88,7 +88,8 @@
       ignoresPersistentStateOnLaunch = "NO"
       debugDocumentVersioning = "YES"
       debugServiceExtension = "internal"
-      allowLocationSimulation = "YES">
+      allowLocationSimulation = "YES"
```
Comment (orta) [inline Artsy.xcodeproj/xcshareddata/xcschemes/Artsy.xcscheme]: oh, definitely don't want this in.

File: Artsy/App/ARAppDelegate.m [hunk 4]
```diff
@@ -92,25 +110,8 @@ - (BOOL)application:(UIApplication *)application willFinishLaunchingWithOptions:
...
-}
-
-
-- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
-{
-    UIApplicationState state = application.applicationState;
-    if (state == UIApplicationStateBackground) {
-        // App is doing a background fetch
-        return YES;
-    }
```
Comment (orta) [inline Artsy/App/ARAppDelegate.m]: So this was causing our bug, if the app is launched by Apple for background fetch, it won't fully set up the app. 

This function isn't used at all in the app now. The work has moved to `applicationWillEnterForeground` which does the UI setup work when we actually will need to show a user interface.

File: Artsy/App/ARAppDelegate.m [hunk 5]
```diff
@@ -130,19 +131,15 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
         // Do not show the splash/onboarding when a user comes in through a user activity, as it breaks the expectation
         // of the user to see the activity. This is probably just an edge-case, most people will probably launch the app
         // after installing it.
-        if (launchOptions[UIApplicationLaunchOptionsUserActivityDictionaryKey] == nil) {
+        if (self.initialLaunchOptions[UIApplicationLaunchOptionsUserActivityDictionaryKey] == nil) {
```
Comment (orta) [inline Artsy/App/ARAppDelegate.m]: trade-off of moving to `applicationWillEnterForeground`is that we don't get an arg of `(NSDictionary *)launchOptions` to the method anymore, so they're kept on the app delegate.

File: Artsy/App/ARAppDelegate.m [hunk 6]
```diff
@@ -130,19 +131,15 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
...
     }
 
-    ARSimpleShowFeedViewController *topVC = (id)ARTopMenuViewController.sharedController.rootNavigationController.topViewController;
     [ArtsyAPI getXappTokenWithCompletion:^(NSString *xappToken, NSDate *expirationDate) {
         // Sync clock with server
         [ARSystemTime sync];
 
-        // Start doing the network calls to grab the feed
-        [topVC refreshFeedItems];
-
```
Comment (orta) [inline Artsy/App/ARAppDelegate.m]: This is now done inside the ShowFeedVC after `viewDidLoad`, this was done to save some networking until needed. However, as it's not needed it's OK.

File: Artsy/Models/Feed_Timelines/ARFeedSubclasses.m [hunk 7]
```diff
@@ -51,14 +51,14 @@ @implementation ARShowFeed
...
     [ArtsyAPI getFeedResultsForShowsWithCursor:cursor pageSize:pageSize success:^(id JSON) {
-        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        ar_dispatch_async(^{
             __strong typeof (wself) sself = wself;
             NSOrderedSet *items = [sself parseItemsFromJSON:JSON];
-            dispatch_async(dispatch_get_main_queue(), ^{
+            ar_dispatch_main_queue(^{
                 success(items);
             });
         });
```
Comment (orta) [inline Artsy/Models/Feed_Timelines/ARFeedSubclasses.m]: so we can run synchronously in tests

File: Artsy/View_Controllers/Util/State_Restoration/UIViewController+ARStateRestoration.h [hunk 8]
```diff
@@ -1,11 +0,0 @@
...
-
-
-@interface UIViewController (ARStateRestoration)
-
-/// Automatic setup of the restoration identifier and class
-/// based on the class of the View Controller instance
-
-- (void)setupRestorationIdentifierAndClass;
-
-@end
```
Comment (orta) [inline Artsy/View_Controllers/Util/State_Restoration/UIViewController+ARStateRestoration.h]: used to be a feature we supported, turned into a total nightmare and removed.

**Comment by orta** [comment]
This is green now

File: Artsy_Tests/Networking_Tests/API_Modules/ARUserManagerTests.m [hunk 9]
```diff
@@ -148,17 +149,9 @@ + (void)clearUserData:(ARUserManager *)manager useStaging:(id)useStaging;
             expect([[NSUserDefaults standardUserDefaults] valueForKey:ARUseStagingDefault]).to.beFalsy();
             expect([[NSUserDefaults standardUserDefaults] valueForKey:@"TestKey"]).to.beNil();
         });
-
```
Comment (orta) [inline Artsy_Tests/Networking_Tests/API_Modules/ARUserManagerTests.m]: it's impossible to get into this state in the app

File: Artsy_Tests/Networking_Tests/API_Modules/ARUserManagerTests.m [hunk 10]
```diff
@@ -176,21 +169,13 @@ + (void)clearUserData:(ARUserManager *)manager useStaging:(id)useStaging;
...
         });
 
-        it(@"does not set staging value passed is nil", ^{
-            expect([[NSUserDefaults standardUserDefaults] valueForKey:ARUseStagingDefault]).to.beNil();
-            expect([[NSUserDefaults standardUserDefaults] valueForKey:@"TestKey"]).to.equal(@"test value");
-            [ARUserManager clearUserData:[ARUserManager sharedManager] useStaging:nil];
-            expect([[NSUserDefaults standardUserDefaults] valueForKey:ARUseStagingDefault]).to.beNil();
-            expect([[NSUserDefaults standardUserDefaults] valueForKey:@"TestKey"]).to.beNil();
-        });
-
```
Comment (orta) [inline Artsy_Tests/Networking_Tests/API_Modules/ARUserManagerTests.m]: same, the change though is due to the `ARAppDelegate`tests. Didn't want to have to refactor all of this stuff to do this https://github.com/artsy/eigen/issues/49

**Comment by orta** [comment]
Not shippable, looks after like testing on device it doesn't load up in an order I expected.

**Comment by orta** [comment]
Fixed and f'pushed. Bug was difference between `applicationDidBecomeActive` and `applicationWillEnterForeground`.

http://www.abdus.me/ios-programming-tips/applicationdidbecomeactive-vs-applicationwillenterforeground/

( basically `applicationWillEnterForeground` means it _has_to have gone into the BG )

**Comment by alloy** [comment]
Had to restart the CI build, but it’s all green now, so merging :+1:

**Pull request merged**